src/lib/wallet.js

```diff
@@ -138,43 +130,35 @@ export default {
         nativeMode: true,
         compilerUrl,
         name: 'Superhero',
-        async onConnection({ info: { icons, name } }, action, origin) {
-          const originUrl = toURL(origin);
+        async onConnection(aepp, action) {
+          // DO SMTH WITH URL
+          const { protocol, host } = getAeppUrl(aepp);
           if (
-            (NO_POPUP_AEPPS.includes(originUrl.hostname) ||
-              (await getAeppAccountPermission(
-                originUrl.hostname,
-                store.state.account.publicKey,
-              ))) &&
-            !(await checkPermissions(action.method))
-          ) {
-            action.accept();
-            return;
-          }
-          try {
-            await store.dispatch('modals/open', {
-              name: 'confirm-connect',
-              app: {
-                name,
-                icons,
-                protocol: originUrl.protocol,
-                host: originUrl.hostname,
-              },
-            });
-            await store.dispatch('setPermissionForAccount', {
-              host: originUrl.hostname,
-              account: store.state.account.publicKey,
-            });
+            !(await requestAddressForAepp(aepp, store.state.account.publicKey, async () =>
+              store.dispatch('modals/open', {
+                name: 'confirm-connect',
+                app: {
+                  name: aepp.name,
+                  icons: aepp.icons,
+                  protocol,
+                  host,
+                },
+              }),
+            ))
+          )
             action.accept();
-          } catch (error) {
-            action.deny();
-            if (error.message !== 'Rejected by user') throw error;
-          }
+          else action.deny();
         },
         onSubscription: acceptCb,
-        onSign: acceptCb,
+        onSign: (aepp, { accept }) => {
+          store.commit('addTxToHost', {
```

Comment:
I can't find where it is used later

Reply:
ok, I got how it works, but I the Base app I think we were handling this a way better